source/acquaman/BioXAS/BioXASMainXASScanActionController.cpp

```diff
@@ -99,13 +99,35 @@ AMAction3* BioXASMainXASScanActionController::createInitializationActions()
     stage4->addSubAction(scaler->createStartAction3(true));
     stage4->addSubAction(scaler->createWaitForDwellFinishedAction(regionTime + 5.0));
 
+    // setup beamline for dark current correction.
+
+    AMListAction3 *darkCurrentSetup = new AMListAction3(new AMListActionInfo3("BioXAS Main Dark Current Setup", "BioXAS Main Dark Current Setup"), AMListAction3::Parallel);
+
+    for (int i = 0, size = configuration_->detectorConfigurations().count(); i < size; i++) {
+	    AMDetector *detector = AMBeamline::bl()->exposedDetectorByInfo(configuration_->detectorConfigurations().at(i));
+	    bool sharedSourceFound = false;
```

Comment:
I think it was originally included so that the scaler performs one dark current measurement for all channel detectors, not one for each channel detector. Where I think I went wrong here was not including a line that changes sharedSourceFound from false to true when the channel detector has a shared trigger source.